### ISSUE TITLE
how to log the Similarly_evaluator_results for triplet loss training. 

### ISSUE THREAD

**Issue by avinashronanki** [UKPLab/sentence-transformers#1378]
Hi author, 
First and foremost, I want to express my sincere thanks for your excellent work.

For triplet loss training, I'm not sure how to log the Similarly evaluator results. I tried the following code and received the following error, with the values 'nan'.
```
# loss function 
train_loss = losses.TripletLoss(model=model)  


evaluator = EmbeddingSimilarityEvaluator.from_input_examples(train_examples)

model.fit(
    train_objectives = [(train_dataloader, train_loss)],
    epochs = 1,
    evaluation_steps=50,
    evaluator=evaluator,
    warmup_steps= int(0.1* len(train_dataloader)),
    output_path=model_save_path )
```
``` 
train_examples = [ 'anchor' , 'positive', 'negative'] 
```
error
`PearsonRConstantInputWarning: An input array is constant; the correlation coefficent is not defined.
SpearmanRConstantInputWarning: An input array is constant; the correlation coefficent is not defined.`

output:
<img width="867" alt="Screenshot 2022-01-16 at 18 07 59" src="https://user-images.githubusercontent.com/47271298/149670867-dcfdc245-ae01-4f06-8bf8-45fd5ed71604.png">

Interestingly, was able to log the loss values for CosineSimilarityLoss
```
train_loss = losses.CosineSimilarityLoss(model)

evaluator = EmbeddingSimilarityEvaluator.from_input_examples(train_examples)

#Tune the model
model.fit(train_objectives=[(train_dataloader, train_loss)],
          evaluation_steps=100,
        evaluator=evaluator,
          epochs=1, warmup_steps=32, output_path=model_save_path)
#save the model 
model = SentenceTransformer(model_save_path) 
```

```
train_examples = ['My first sentence', 'My second sentence', '0.8']
```
output:
```
2022-01-10 14:02:42 - EmbeddingSimilarityEvaluator: Evaluating the model on  dataset in epoch 0 after 50 steps:
2022-01-10 14:02:55 - Cosine-Similarity :	Pearson: 0.9975	Spearman: 0.8660
2022-01-10 14:02:55 - Manhattan-Distance:	Pearson: 0.9775	Spearman: 0.8660
2022-01-10 14:02:55 - Euclidean-Distance:	Pearson: 0.9817	Spearman: 0.8660
2022-01-10 14:02:55 - Dot-Product-Similarity:	Pearson: 0.9543	Spearman: 0.8654

2022-01-10 14:03:21 - EmbeddingSimilarityEvaluator: Evaluating the model on  dataset in epoch 0 after 100 steps:
2022-01-10 14:03:34 - Cosine-Similarity :	Pearson: 0.9952	Spearman: 0.8660
2022-01-10 14:03:34 - Manhattan-Distance:	Pearson: 0.9740	Spearman: 0.8660
2022-01-10 14:03:34 - Euclidean-Distance:	Pearson: 0.9787	Spearman: 0.8660
2022-01-10 14:03:34 - Dot-Product-Similarity:	Pearson: 0.9488	Spearman: 0.8656
```

1. What is the best way to keep track of the similarity evaluation scores for the triplet loss throughout training?
2. Is it possible that I'm using the incorrect input? (as the negative is a string value )

can I get some advice on it?

Thank you

**Comment by imen93** [comment]
You should use the TripletEvaluator instead of EmbeddingSimilarityEvaluator.  https://www.sbert.net/docs/package_reference/evaluation.html
